oauth/src/OAuth/Database.hs

```diff
@@ -0,0 +1,44 @@
+module OAuth.Database
+  ( Database
+  , loadSession
+  , saveGmailEmail
+  , saveSession
+  )
+where
+
+import OAuth.Prelude
+import OAuth.Session
+
+import qualified Data.Aeson                as JSON
+import qualified Data.ByteString.Lazy      as LBS
+import qualified Data.ByteString.Lazy.UTF8 as BLU
+import qualified Data.Text                 as Text
+import qualified System.IO                 as IO
+
+type Database = Session
+
+loadSession :: Text -> Text -> RIO env (Either Text Session)
+loadSession _serviceName _accountId = do
+  database <- liftIO $ JSON.eitherDecode <$> LBS.readFile sessionFilename
+
+  pure $ case database of
+    Left  msg -> Left $ Text.pack msg
+    Right db  -> Right db
+
+saveGmailEmail :: Text -> Text -> LBS.ByteString -> IO ()
+saveGmailEmail accountId messageId json =
+  IO.writeFile filepath $ BLU.toString json
+  where
+    filepath :: IO.FilePath
+    filepath = "data/emails/gmail/" <> filename
+
+    filename :: String
+    filename = Text.unpack $ accountId <> "-" <> messageId <> ".json"
+
+saveSession :: Session -> RIO env Session
+saveSession session = do
+  liftIO $ IO.writeFile sessionFilename (BLU.toString $ JSON.encode session)
```

Comment:
@snusnu are the functions in this module meant for debugging in a non-production environment ?

Reply:
@epicallan Exactly. IMO those (and other code) should still be removed before merge.

Reply:
> IMO those (and other code) should still be removed before merge.

@snusnu I have possibly removed all the code we don't need. Take a look and let me know if there is anything I may have left. I have also made some other minor changes.

Reply:
@epicallan Afaict, what's left seems to be enough. Anything we may eventually need in top of that, can be added once the need for it arises.

Reply:
Alright, thanks for checking.